### PR TITLE
React bootstrap v2, React 18, Redux and Misc. Npm Packages Upgrade

### PR DESCRIPTION
**Pull request by Onurcankaratay**
This pull request primarily addresses the transition from React 17 to React 18, while also including updates for Redux, Auth0, and other third-party integrations heavily used in the portal, ensuring compatibility with React 18.

Ticket:  [Trello 1](https://trello.com/c/eaJhXgzJ/345-react-bootstrap-v2-upgrade-for-smaht) & [Trello 2](https://trello.com/c/vl9gUppW/344-fix-deprecated-scss-features)
Related PR:  [SPC](https://github.com/4dn-dcic/shared-portal-components/pull/261)

### Updates
- **React**: Upgraded from v17 to v18
- **Redux**: Updated from v4 to v5
- **Auth0-Lock**: Upgraded from v11 to v12
- **Gulp.js**: Updated from v4 to v5

Additionally, this pull request implements the migration to Bootstrap v5 and updates to React-Bootstrap v2, ensuring that styles and components are fully compatible with the latest standards.

- **Bootstrap v5 Migration**: Adapted styles and class names to align with Bootstrap 5.
- **React-Bootstrap v2 Update**: Updated components and class names to conform to React-Bootstrap 2 standards.
- **FontAwesome v6 Integration**: Upgraded to FontAwesome v6 to resolve console SCSS build warnings and ensure future compatibility.

**Note**: React-Bootstrap v2 encompasses React 18, so it can be disregarded. [PR](https://github.com/4dn-dcic/shared-portal-components/pull/259)

### CODE REVIEW

**Comment by utku-ozturk** [comment]
[Fix deprecated scss features](https://trello.com/c/vl9gUppW/344-fix-deprecated-scss-features)

**Comment by utku-ozturk** [comment]
[React 18 and react-bootstrap v2 upgrades for SMaHT](https://trello.com/c/eaJhXgzJ/345-react-18-and-react-bootstrap-v2-upgrades-for-smaht)

**Review by Bianca-Morris — COMMENTED** [review]
Looking pretty good. For the most part, I haven't noticed any major issues. Just a few small things.

A few notes I didn't know where else to put:

- Locally, I'm noticing some odd behavior on `/health` where the tooltips loaded after pressing the button are not working. Doesn't seem to be happening on wolf, though.

[Image from Gyazo](https://gyazo.com/21768104ffdd31dfbddb3f1deff70427)

- There are a few other static pages on prod that you might not have on local/wolf that you should see if you can pull and double check (or look at carefully once this gets to staging): `/bam-qc-overview` and `/submission-status`
    - On wolf, the `/submission-status` page seems to be broken. Not sure if that's related to broken data on wolf, though, or something with this PR.
    [Image from Gyazo](https://gyazo.com/eaa9b92dbf7b8d9d2db7d068af047ef1)

File: src/encoded/static/scss/encoded/_bootstrap-theme-overrides.scss [hunk 1]
```diff
@@ -116,20 +99,24 @@ $margin-util-sizes-mini: 1, 2, 3, 4, 5, 6, 7, 8; 		// * 1px
 }
 
 // Note: Temporary; use btn-sm styling for buttons on tablet and mobile
-@include media-breakpoint-down(xs){
+@media (max-width: 576px) {
```
[the remark this answers is not in the document]
Comment (Bianca-Morris) [reply src/encoded/static/scss/encoded/_bootstrap-theme-overrides.scss:102]: Actually, yeah it looks like we have a `$grid-breakpoint` `sm` at 576px. Maybe one of the `$container-max-width`s could be adjusted/added? IDK if this is even worth doing honestly. Your call.

**Comment by crfmc** [comment]
Hi @Onurcankaratay! I noticed some styles are now being overwritten. I added screenshots comparing this branch with the current style. Very sorry for the poorly formatted comment!


## Here the border radius top-left and top-right should be 0.25rem, and for some reason there is a gap below now. I'm not sure why. The badge color is also changed.

**This branch:**
Screenshot 2024-10-04 at 2 19 16 PM

**Current style:**
Screenshot 2024-10-04 at 2 20 32 PM

---

<br>
<br>

## The background of these popover headers should be a little lighter:

__This branch: Homepage, the /about/consortium/awardees page, and the /about.consortium/data page__

Screenshot 2024-10-04 at 2 28 41 PM
Screenshot 2024-10-04 at 2 36 40 PM
Screenshot 2024-10-04 at 2 39 03 PM


**Current style:**
Screenshot 2024-10-04 at 2 27 42 PM

---

<br>
<br>

## These tables have their background color overwritten:
Screenshot 2024-10-04 at 2 33 51 PM
Screenshot 2024-10-04 at 2 40 26 PM
Screenshot 2024-10-04 at 2 44 08 PM

**Comment by crfmc** [comment]
Hi @Onurcankaratay, thank you for making the changes, they look great! Sorry I missed these small changes before:
- In the /about/consortium/awardees page, the text in the table should be: #212529 but is currently black.
- The table cells in the Access Keys table in the User page looks like it lost some padding, I attached screenshots below.

@Bianca-Morris It might be good to have Shannon take a look in case we missed something?

**This branch:**

Screenshot 2024-10-07 at 9 28 32 AM


**Current:**

Screenshot 2024-10-07 at 9 28 18 AM

**Comment by Onurcankaratay** [comment]
@Bianca-Morris, @crfmc Thanks for your detailed review. I have fixed/refactored the issues you mentioned. Could you have a look when you have a chance please - thanks.

**Comment by Bianca-Morris** [comment]
> @Bianca-Morris, @crfmc Thanks for your detailed review. I have fixed/refactored the issues you mentioned. Could you have a look when you have a chance please - thanks.

I did look at this a small bit, and I think most of my concerns were addressed. But haven't gotten a chance to do a full walkthrough/test again, so don't want to explicitly approve.

I'll be out until next week, so if Cesar and Shannon have an opportunity to look at this and approve the final edits before then, feel free to merge it without my approval. Otherwise, if there are additional things they catch that need to be updated, I'm happy to re-review upon my return.

Thanks for your work on this!

**Comment by crfmc** [comment]
Hi @Onurcankaratay! The only difference I noticed was that the table items on the `/docs/additional-resources/sample-file-nomenclature` page are centered. This is left aligned on the live site. 

Otherwise, everything looks good! Thank you for addressing all of these concerns.

Screenshot 2024-10-16 at 9 16 43 AM

**Comment by Onurcankaratay** [comment]
Hi @crfmc! It looks like there was an indexing issue. I reindexed the page, and it should be okay now.

Screenshot from 2024-10-16 17-39-08

**Pull request merged**